### DISCUSSION TITLE
SMCB authentication flow / Step 4. does not return session_state and code

### DISCUSSION THREAD

**Discussion by fue-medatixx** [Digitaler-Impfnachweis/certification-apis#34] [Q&A] [answered]
Step 4. does not return session_state and code in location header value.
Only auth_session_id is given

HTTP/1.1 302 Found
Content-Length: 0
Date: Tue, 25 May 2021 15:01:57 GMT
Location: https://id.certify.demo.ubirch.com/auth/realms/ubirch-certify/login-actions/required-action?execution=VERIFY_EMAIL&client_id=user-access-ti&tab_id=A097Wp6Q0vg&auth_session_id=36487b52-b55d-4d93-8b8a-e0641b4fe4ce

Therefore the Step 5 fails.
HTTP/1.1 400 Bad Request
Content-Length: 62
Content-Type: application/json
Date: Tue, 25 May 2021 15:01:58 GMT
{"error":"invalid_grant","error_description":"Code not valid"}

**Comment by mge-medisoftware** [comment, marked as answer]
We can confirm this, no session_state and code. 

HTTP/1.1 302 Found
Cache-Control: no-store, must-revalidate, max-age=0
Content-Length: 0
Date: Wed, 26 May 2021 08:29:32 GMT
Location: https://id.certify.demo.ubirch.com/auth/realms/ubirch-certify/login-actions/required-action?execution=VERIFY_EMAIL&client_id=user-access-ti&tab_id=Jr7kOGovCJM&auth_session_id=d4f76443-5655-4ffe-a35d-3c19e4c99e06
Referrer-Policy: no-referrer
Server: istio-envoy
Set-Cookie: KEYCLOAK_LOCALE=; Version=1; Comment=Expiring cookie; Expires=Thu, 01-Jan-1970 00:00:10 GMT; Max-Age=0; Path=/auth/realms/ubirch-certify/; HttpOnly
Strict-Transport-Security: max-age=15552000; includeSubDomains; preload
X-Content-Type-Options: nosniff
X-Envoy-Decorator-Operation: keycloak-certificato-headless.ubirch-certify-demo.svc.cluster.local:80/*
X-Envoy-Upstream-Service-Time: 21
X-Xss-Protection: 1; mode=block

**Reply by fue-medatixx** [reply]
Hello,
the Auth-Flow RU works now. We got an access token back from https://id.ru.impfnachweis.info

**Comment by oliver-steinbrecher** [comment]
We had some environmental issues yesterday .
